core/src/main/java/org/elasticsearch/search/aggregations/bucket/children/ParentToChildrenAggregator.java

```diff
@@ -217,5 +231,33 @@ protected Aggregator doCreateInternal(ValuesSource.Bytes.WithOrdinals.ParentChil
                     valuesSource, maxOrd, pipelineAggregators, metaData);
         }
 
+        private void resolveConfig(AggregationContext aggregationContext) {
+            config = new ValuesSourceConfig<>(ValuesSource.Bytes.WithOrdinals.ParentChild.class);
+            DocumentMapper childDocMapper = aggregationContext.searchContext().mapperService().documentMapper(childType);
+
+            if (childDocMapper != null) {
+                ParentFieldMapper parentFieldMapper = childDocMapper.parentFieldMapper();
+                if (!parentFieldMapper.active()) {
+                    throw new SearchParseException(aggregationContext.searchContext(),
+                            "[children] no [_parent] field not configured that points to a parent type", null); // NOCOMMIT fix exception args
+                }
+                parentType = parentFieldMapper.type();
+                DocumentMapper parentDocMapper = aggregationContext.searchContext().mapperService().documentMapper(parentType);
+                if (parentDocMapper != null) {
+                    // TODO: use the query API
```

Comment:
this comment is not relevant anymore as it seems to be using the query API already?